### PR TITLE
Update reading view fetcher

### PR DESCRIPTION
**Pull request by osamasayed**

### CODE REVIEW

File: src/components/QuranReader/ReadingView/Page.tsx [hunk 1]
```diff
@@ -23,7 +21,10 @@ type PageProps = {
 };
 
 const Page = ({ verses, page, quranReaderStyles, pageIndex }: PageProps) => {
```
Comment (tifa2UP) [inline src/components/QuranReader/ReadingView/Page.tsx:23]: pageNumber

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/\[chapterId\]/\[verseId\]**](https://quran-com-kwxczrfml-qurancom.vercel.app/[chapterId]/[verseId]) | [**/\[chapterId\]**](https://quran-com-kwxczrfml-qurancom.vercel.app/[chapterId]) |
|:-:|:-:|
|<a href="https://quran-com-kwxczrfml-qurancom.vercel.app/[chapterId]/[verseId]">Screenshot of /[chapterId]/[verseId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/1Kafk5hdW0FJJHCxpm87qfoVrTOLHY_y1TrlkmP519yVuleB1woNuufgoie29v5wQgNBldGjRRhKIz1_KpC3qA.png">(view full size)</a> |<a href="https://quran-com-kwxczrfml-qurancom.vercel.app/[chapterId]">Screenshot of /[chapterId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/V8QNLaWemgjZKUSYvuukYdaThnKPfaUUP-pnx7jAL4hIxPNw9elX6dMnDfi1WVQdjbOKBVlRF8MUDy4_WI_Mhw.png">(view full size)</a> |



| [**/juz/\[juzId\]**](https://quran-com-kwxczrfml-qurancom.vercel.app/juz/[juzId]) | [**/page/\[pageId\]**](https://quran-com-kwxczrfml-qurancom.vercel.app/page/[pageId]) |
|:-:|:-:|
|<a href="https://quran-com-kwxczrfml-qurancom.vercel.app/juz/[juzId]">Screenshot of /juz/[juzId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/LiOIvGlCv7_Fdmzv67eJ3zETHZ70Xejm9oEPqZ8RXjLWwvn5aUHV3jezpGYWhjLjmGFlzkZlqoifV4WU1QXOEw.png">(view full size)</a> |<a href="https://quran-com-kwxczrfml-qurancom.vercel.app/page/[pageId]">Screenshot of /page/[pageId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/vyVorchxwTvlrFz5r0KqI7O_p68sOiFlSJx2_NqEsPcaBuCrYFUBesfVxYLBpVkj2mzLOfZ5VKPA4cqBEzqiUw.png">(view full size)</a> |







Commit 26d4b92ba622a904ed477db2641d43cff5316918 (https://quran-com-kwxczrfml-qurancom.vercel.app).

File: src/components/QuranReader/ReadingView/index.tsx [hunk 2]
```diff
@@ -110,44 +83,20 @@ const ReadingView = ({
...
-        const lastRenderedMushafPage =
-          initialFirstMushafPage + renderedPages[renderedPages.length - 1].index;
-        if (lastRenderedMushafPage !== prevMushafPage) {
-          return lastRenderedMushafPage;
-        }
-        return prevMushafPage;
-      });
-    }
-  };
+  const itemContentRenderer = (currentPageIndex: number) => (
```
Comment (tifa2UP) [inline src/components/QuranReader/ReadingView/index.tsx:87]: pageIndex

**Review by tifa2UP — APPROVED** [review]

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/\[chapterId\]/\[verseId\]**](https://quran-com-avmg9h3ii-qurancom.vercel.app/[chapterId]/[verseId]) | [**/\[chapterId\]**](https://quran-com-avmg9h3ii-qurancom.vercel.app/[chapterId]) |
|:-:|:-:|
|<a href="https://quran-com-avmg9h3ii-qurancom.vercel.app/[chapterId]/[verseId]">Screenshot of /[chapterId]/[verseId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/g8GVU3jJTXRYJjHRgXbfNizYV-_gZjRYJgKdE3CLHckTztZ7XeKM7O8wPzeeABZrHAkJn0KTd0RrssiEod7DjQ.png">(view full size)</a> |<a href="https://quran-com-avmg9h3ii-qurancom.vercel.app/[chapterId]">Screenshot of /[chapterId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/31Cl76azugtCu6Wxybh5LHzmGUywi3uWihbZKiUqjCxXI6VDnTpEZGoC2_Cf_761Ok6kj4Fdq5WCMm05bPLT_g.png">(view full size)</a> |



| [**/juz/\[juzId\]**](https://quran-com-avmg9h3ii-qurancom.vercel.app/juz/[juzId]) | [**/page/\[pageId\]**](https://quran-com-avmg9h3ii-qurancom.vercel.app/page/[pageId]) |
|:-:|:-:|
|<a href="https://quran-com-avmg9h3ii-qurancom.vercel.app/juz/[juzId]">Screenshot of /juz/[juzId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/9qgN1ubj9pZ75VZVAG7YiSqHbwd6FrGEFnvGNmFsJstQdA3UzVFDcJQU-5Bu6i73TshXUT5d3VDmBi9LRUveWg.png">(view full size)</a> |<a href="https://quran-com-avmg9h3ii-qurancom.vercel.app/page/[pageId]">Screenshot of /page/[pageId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/XR3uC90luQEyTS2M5BuWocimsE9G8TelPza3OtkQ2Rp4rciPRI_rpovcZlVGTs5UUYZnCIDJgHd8gyJ2xXqEeA.png">(view full size)</a> |







Commit c61c0611cef749445cc91a7a943a18e11ab88638 (https://quran-com-avmg9h3ii-qurancom.vercel.app).

**Pull request merged**